### ISSUE TITLE
Training with Dipole Model

### ISSUE THREAD

**Issue by mberrens** [ACEsuit/mace#710]
I am attempting to train a dipole+energy+force model with the following input script:

`python ${mace_scripts}/run_train.py \
  --name="MACE_model" \
  --train_file="train_example.extxyz" \
  --valid_fraction=0.1 \
  --E0s='average' \
  --model="EnergyDipolesMACE" \
  --hidden_irreps='128x0e + 128x1o' \
  --r_max=6.0 \
  --error_table="EnergyDipoleRMSE"\
  --energy_key="REF_energies"\
  --forces_key="REF_forces"\
  --dipole_key="dipoles"\
  --batch_size=1 \
  --loss="energy_forces_dipole"\
  --valid_batch_size=1 \
  --max_num_epochs=150 \
  --ema \
  --ema_decay=0.99 \
  --amsgrad \
  --default_dtype="float32" \
  --device=cuda \
  --seed=123`

 and following header for the training data:

`366
Lattice="12.562 0.0 0.0 0.0 13.055 0.0 0.0 0.0 25.936638502076306" Properties=species:S:1:pos:R:3:REF_forces:R:3:dipoles:R:3 REF_energies=-59355.946217 overall_charge=0.0 pbc="T T T"`

However, I have attempted many variations of the dipole key ('REF_dipole', 'REF_dipoles', 'dipoles', 'dipole') but MACE doesn't seem to accept any of these keys and ignores the input atomic dipole values of the training data. Such that I see a similar issue to that seen in a previous ([bug](https://github.com/ACEsuit/mace/issues/412)), where I see an unreasonably large rel MU RMSE %. I believe that the issue is MACE isn't recognizing the dipole key because I used the same input script with training data that had no atomic dipoles in the xyz files and got the same exact result as with the dipoles in the xyz file. 

Below is a portion of the output log:

2024-11-21 15:56:51.676 INFO: Epoch 147: head: default, loss=  0.1478, RMSE_E_per_atom=     3.6 meV, RMSE_F=    38.4 meV / A, RMSE_Mu_per_atom=    0.11 mDebye
2024-11-21 15:56:58.111 INFO: Epoch 148: head: default, loss=  0.1473, RMSE_E_per_atom=     4.2 meV, RMSE_F=    38.4 meV / A, RMSE_Mu_per_atom=    0.11 mDebye
2024-11-21 15:57:04.480 INFO: Epoch 149: head: default, loss=  0.1472, RMSE_E_per_atom=     3.8 meV, RMSE_F=    38.4 meV / A, RMSE_Mu_per_atom=    0.11 mDebye
2024-11-21 15:57:04.562 INFO: Training complete
2024-11-21 15:57:04.562 INFO:
2024-11-21 15:57:04.562 INFO: ===========RESULTS===========
2024-11-21 15:57:04.562 INFO: Computing metrics for training, validation, and test sets
2024-11-21 15:57:04.565 INFO: Loading checkpoint: ./checkpoints/MACE_model_run-123_epoch-149.pt
2024-11-21 15:57:04.658 INFO: Loaded Stage one model from epoch 149 for evaluation
2024-11-21 15:57:04.659 INFO: Evaluating train_default ...
2024-11-21 15:57:06.811 INFO: Evaluating valid_default ...
2024-11-21 15:57:07.168 INFO: Error-table on TRAIN and VALID:
+---------------+---------------------+------------------+--------------+-------------------------+---------------+
|  config_type  | RMSE E / meV / atom | RMSE F / meV / A | rel F RMSE % | RMSE MU / mDebye / atom | rel MU RMSE % |
+---------------+---------------------+------------------+--------------+-------------------------+---------------+
| train_default |            3.1      |         27.7     |        2.9   |              0.1        |  3023599274.5 |
| valid_default |            3.8      |         38.4     |        4.1   |              0.1        |  4179392382.5 |
+---------------+---------------------+------------------+--------------+-------------------------+---------------+
2024-11-21 15:57:07.168 INFO: Saving model to checkpoints/MACE_model_run-123.model
2024-11-21 15:57:07.395 INFO: Compiling model, saving metadata to MACE_model_compiled.model
2024-11-21 15:57:09.154 INFO: Done

Is there a key value for the dipoles that I should use instead?

**Comment by ilyes319** [comment]
hey, when you changed the input key, did you make sure to also change it in the xyz file? 
I don't "dipole" would work, you to rename everything to "REF_dipole" both in xyz and in the input dipole_key.

**Comment by gabor1** [comment]
just so you know, the dipole prediction is just a separate output at this point, it is not integrated with the energy in any way

**Comment by mberrens** [comment]
Yes I made sure to change the key in both the xyz and input dipole key each time. I tried 'REF_dipole" and "REF_dipoles". Is there an example I could see of training a 'EnergyDipolesMACE' model with training data that has the atomic dipoles so I could see how that is formatted? Because neither REF_dipole or REF_dipoles worked for me.

And I'm not sure what you mean it is a separate output? I thought the point of the 'EnergyDipolesMACE model versus the 'AtomicDipoleMACE' model is that with the 'EnergyDipolesMACE' model you can train simultaneously on energies and the dipole moments

**Comment by gabor1** [comment]
what I mean is that the energy of these dipoles is not computed/used/fitted. it's simply an output prediction. Full electrostatic mace is coming (with variable charges and dipoles).

**Comment by mberrens** [comment]
Thank you for the clarification. I understand that it is just an extra output. I believe I figured out why MACE wasn't recognizing the key for me, this is because it seems that both the EnergyDipolesModel and the AtomicDipolesModel only predict the total dipole moment of the system and not the atomic dipole moments, is this correct? 

I am trying to predict the atomic dipoles (just as an extra output) as the model titles suggest and I believe I am having a formatting issue which is why I was hoping I could see an example? Unless it is not currently set up to have an extra output of atomic dipoles and only total dipole moment of the system. 

You can see below, when I try to train on atomic dipoles with the 'AtomicDipolesModel' I get a 0.0 RMSE 
**Header**
Lattice="12.562 0.0 0.0 0.0 13.055 0.0 0.0 0.0 25.936638502076306" Properties=species:S:1:pos:R:3:REF_forces:R:3:REF_dipoles:R:3 REF_energies=-59355.946217 overall_charge=0.0 pbc="T T T"

**Result**
+---------------+-------------------------+--------------------+
|  config_type  | RMSE MU / mDebye / atom | relative MU RMSE % |
+---------------+-------------------------+--------------------+
| train_default |             0.00        |           0.0      |
| valid_default |             0.00        |           0.0      |
+---------------+-------------------------+--------------------+

but when I train on total dipole of the system with the 'AtomicDipolesModel' I get a result as I would expect

**Header**
Lattice="12.562 0.0 0.0 0.0 13.055 0.0 0.0 0.0 25.936638502076306" Properties=species:S:1:pos:R:3:REF_forces:R:3 REF_energies=-59355.946217 REF_dipoles="-24.156934 24.348011 -1.246111" overall_charge=0.0 pbc="T T T"

**Result**
2024-11-27 11:02:36.458 INFO: Error-table on TRAIN and VALID:
+---------------+-------------------------+--------------------+
|  config_type  | RMSE MU / mDebye / atom | relative MU RMSE % |
+---------------+-------------------------+--------------------+
| train_default |             1.51        |           1.9      |
| valid_default |             1.03        |           1.4      |
+---------------+-------------------------+--------------------+

**Comment by mberrens** [comment]
Okay I see - thank you for the clarification
